src/Symfony/Component/HttpKernel/DependencyInjection/AddClassesToCachePass.php

```diff
@@ -36,12 +36,20 @@ public function __construct(Kernel $kernel)
      */
     public function process(ContainerBuilder $container)
     {
-        $classes = array();
-        $annotatedClasses = array();
+        $classes = $this->kernel->getClassesToCompile();
+        $annotatedClasses = $this->kernel->getAnnotatedClassesToCompile();
+
         foreach ($container->getExtensions() as $extension) {
-            if ($extension instanceof Extension) {
-                $classes = array_merge($classes, $extension->getClassesToCompile());
-                $annotatedClasses = array_merge($annotatedClasses, $extension->getAnnotatedClassesToCompile());
+            if (!$extension instanceof Extension) {
+                continue;
+            }
+
+            foreach ($extension->getClassesToCompile() as $class) {
+                $classes[] = $class;
+            }
+
+            foreach ($extension->getAnnotatedClassesToCompile() as $annotatedClass) {
+                $annotatedClasses[] = $annotatedClass;
```

Comment:
Basically due to performance. Array merge in loops is slow (at least in php5, haven't benchmarked 7 myself): https://github.com/dseguy/clearPHP/blob/master/rules/no-array_merge-in-loop.md